modules/collection/programs/fish.nix

```diff
@@ -0,0 +1,98 @@
+{
+  pkgs,
+  lib,
+  config,
+  ...
+}: let
+  inherit (lib.options) mkOption mkEnableOption mkPackageOption;
+  inherit (lib.strings) typeOf concatMapAttrsStringSep concatMapStringsSep splitString;
+  inherit (lib.modules) mkIf;
+  inherit (lib.types) either str path oneOf attrsOf nullOr bool;
+  inherit (lib.attrsets) mapAttrs' nameValuePair isDerivation;
+
+  cfg = config.rum.programs.fish;
+  env = config.environment.sessionVariables;
+
+  toFish = strOrPath: fileName:
+    if (typeOf strOrPath) == "string"
+    then pkgs.writers.writeFish fileName strOrPath
+    else if isDerivation strOrPath
+    then strOrPath
+    else throw "Input is of invalid type ${typeOf strOrPath}, expected `path` or `string`.";
+
+  toFishFunc = strOrPath: funcName:
+    toFish (
+      if (typeOf strOrPath) == "string"
+      then ''
+        function ${funcName};
+        ${concatMapStringsSep "\n" (line: "\t${line}") (splitString "\n" strOrPath)}
+        end
+      ''
+      else strOrPath
+    ) "${funcName}.fish";
+in {
+  options.rum.programs.fish = {
+    enable = mkEnableOption "fish";
+
+    package = mkPackageOption pkgs "fish" {};
+
+    config = mkOption {
+      default = null;
+      type = nullOr (either str path);
+      description = ''
+        The main configuration for fish, written verbatim to `.config/fish/config.fish`.
+      '';
+    };
+
+    exportEnvVars = mkOption {
+      default = true;
+      type = bool;
+      description = ''
+        Wether to export environment variables set in ''${config.environment.systemVariables}.
+      '';
+    };
```

Comment:
Should this even be an option? Do you know/have any use-cases in which you might not want to import your environment variables?

Reply:
Doesn't make a lot of sense, considering if this is disabled, fish.enable does _nothing_.

Reply:
I meant removing the option and just importing environment variables unconditionally

Reply:
Yeah that's what I meant

Reply:
I'm not sure I follow. `fish.enable` does do something..? It adds the package to user environment, it's just that unconfigured it will not do anything else except add the package to `hjem.users.<user>.packages` as well *should* setup environment variables so a usable shell can be made available.

Reply:
Yeah, dw my brain is fried. If there ever is someone that wants the env vars to be removable we can add it then